dynamiqs/solvers/ode/adaptive_solver.py

```diff
@@ -211,6 +199,16 @@ def integrate_augmented(
                 cache = (dt, error)
                 dt = t1 - t
 
+            # the computation graph attached to `y` and `a` is automatically freed after
+            # next line, because there are no more references to the original tensors
+            # (the old `y`, `a`, `ft` and `lt` go out of scope)
+            y, a, ft, lt = (
+                new_leaf_tensor(y),
+                new_leaf_tensor(a),
+                new_leaf_tensor(ft),
+                new_leaf_tensor(lt),
+            )
```

Comment:
I was curious about your opinion haha. Took me a while to understand how the subgraphs attached to obsolete tensors were freed with the old code, I thought it would be good to make it as clear as possible. 😊